packages/api/tests/endpoints/test-granules-updateGranulesAndFilesCollection.js

```diff
@@ -0,0 +1,339 @@
+        path: t.context.collectionId + '/' + pgGranule.granule_id,
+      })
+    );
+  }
+
+  t.context.pgFiles = await t.context.filePgModel.insert(knex, t.context.files);
+  t.context.apiGranules = [];
+
+  await esClient.client.indices.refresh({ index: t.context.esIndex });
+  await Promise.all(t.context.granules.map(async (granule) => {
+    const newGranule = await translatePostgresGranuleResultToApiGranule(knex, {
+      ...granule,
+      collectionName: t.context.collection.name,
+      collectionVersion: t.context.collection.version,
+    });
+    t.context.apiGranules.push(newGranule);
+    await indexer.indexGranule(esClient, newGranule, t.context.esIndex);
+    await esClient.client.indices.refresh({ index: t.context.esIndex });
+  }));
+
+  // update all of the granules to be moved to the new collection
+  t.context.movedGranules.push(await simulateGranuleUpdate(knex, t.context.granules,
+    t.context.collection, t.context.collectionId, t.context.collectionId2));
+
+  t.context.movedGranules = t.context.movedGranules.flat();
+});
+
+test.after.always(async (t) => {
+  await accessTokenModel.deleteTable();
+  await recursivelyDeleteS3Bucket(process.env.system_bucket);
+  await cleanupTestIndex(t.context);
+  await destroyLocalTestDb({
+    knex: t.context.knex,
+    knexAdmin: t.context.knexAdmin,
+    testDbName,
+  });
+});
+
+test.serial('PATCH /granules/batchRecords successfully updates granules to new collectionId in PG and ES', async (t) => {
+  const {
+    granuleIds,
+    granulePgModel,
+    apiGranules,
+    collectionId2,
+    collection2,
+    collectionCumulusId2,
+    esIndex,
+    esClient,
+    knex,
+  } = t.context;
+
+  const params = {
+    apiGranules: apiGranules,
+    collectionId: collectionId2,
+  };
+
+  const response = await request(app)
+    .patch('/granules/batchRecords')
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(params)
+    .expect(200);
+
+  const { message } = response.body;
+  t.true(message.includes('Successfully wrote granules'));
+  const returnedGranules = await Promise.all(granuleIds.map((id) =>
+    getUniqueGranuleByGranuleId(knex, id, granulePgModel)));
+
+  for (const granule of returnedGranules) {
+    t.is(granule.collection_cumulus_id, collectionCumulusId2);
+    const apiGranule = await translatePostgresGranuleResultToApiGranule(knex, {
+      ...granule,
+      collectionName: collection2.name,
+      collectionVersion: collection2.version,
+    });
+    const esGranule = await esClient.client.get({
+      index: esIndex,
+      type: 'granule',
+      id: granule.granule_id,
+      parent: apiGranule.collectionId,
+    }).then((res) => res.body);
+
+    t.is(apiGranule.collectionId, collectionId2);
+    t.is(esGranule._source.collectionId, collectionId2);
+  }
+});
+
+test.serial('PATCH /granules/batchPatch successfully updates a batch of granules', async (t) => {
```

Comment:
We aren't testing batch updates of other granule fields here and/or that concurrency options are properly handled - should we?

Reply:
I think it's worth thinking about, particularly as the code evolves  - but at a minimum we should ensure the endpoint specific params are tested.